src/codelabs/null-safety.md

```diff
@@ -348,23 +352,24 @@ as providing an alternative value if the left-hand side is `null`.
 
 ## Type promotion
 
-With sound null safety, Dart's
-[flow analysis](/null-safety/understanding-null-safety#flow-analysis) has
-been extended to take nullability into account. Nullable variables that
-can't possibly contain null values are treated like non-nullable variables.
-This behavior is called
+With sound null safety, we extended Dart's
```

Comment:
I'd drop the history part of this (i.e. change "we extended Dart's flow analysis to account for..." to "Dart's flow analysis accounts for...")